numpy/core/arrayprint.py

```diff
@@ -273,6 +273,40 @@ def get_printoptions():
     return _format_options.copy()
 
 
+@contextlib.contextmanager
+def printoptions(*args, **kwds):
+    """Context manager for setting print options.
+
+    See `set_printoptions` for the full description of available options.
+
+    Examples
+    --------
+
+    `printoptions` sets print options temporarily, and restores them back
+    at the end of the `with`-block:
```

Comment:
I think this belongs in the main doc section above this heading. Maybe the one-line summary should be

> A context manager that sets print options for the scope of the `with` block, restoring the old options at the end